### PR TITLE
Change OSBS extra to copy to directory and add optional ability to ad…

### LINKED ISSUE #708: Add optional copy to extra_dir functionality
Spawned from #684 

We should add the ability to add copy to https://docs.cekit.io/en/latest/descriptor/image.html#osbs-extra-directory e.g. `copy_extra_dir_to` to denote a target location in the image to copy the files to.

To make this easier we should change how `extra_dir` works and copy the **directory** across and not just the contents. This is a user visible change.

This will make it easier to manage and add the `COPY` instructions. It would also then make it possible to implement https://github.com/cekit/cekit/issues/700 as there is a known location to manage.

cc @AntStephenson @tchughesiv

### PR DESCRIPTION
**Pull request by rnc**
…d COPY instruction

Fixes #708

### CODE REVIEW

**Comment by tchughesiv** [comment]
tested w/ these changes and this branch... seems to do that job!
https://github.com/kiegroup/kie-cloud-operator/pull/623

File: cekit/template_helper.py [hunk 1]
```diff
@@ -88,6 +88,13 @@ def cachito(self, image):
         if image.get('osbs', {}).get('configuration', {}).get('container', {}).get('remote_source'):
             return True
 
+    def extra_dir(self, image):
+        return image.get('osbs', {}).get('extra_dir', {'osbs_extra'})
```
Comment (goldmann) [inline cekit/template_helper.py:92]: Why `{}` around `osbs_extra`?

File: cekit/template_helper.py [hunk 2]
```diff
@@ -88,6 +88,13 @@ def cachito(self, image):
...
+
+    def extra_dir_target(self, image):
+        if image.get('osbs', {}).get('extra_dir_target', {}):
+            return image.get('osbs', {}).get('extra_dir_target', {})
```
Comment (goldmann) [inline cekit/template_helper.py:96]: Why not just `return image.get('osbs', {}).get('extra_dir_target', {})` and removing the `if` entirely?

Comment (goldmann) [reply cekit/template_helper.py:96]: Why the default value for `extra_dir_target` is set to `{}`?

File: docs/descriptor/includes/image/osbs.rst [hunk 3]
```diff
@@ -50,6 +49,31 @@ in the :ref:`generation phase<handbook/building/build-process:Generating require
...
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Key
+    ``extra_dir_target``
+Required
+    No
+
+If this key is set then the contents of ``extra_dir`` will be copied during the image build to the location specified by this key. If this is *not* set, no ``COPY`` commands will be added. For example, with the value of ``/`` then
+
+.. code-block:: bash
```
Comment (goldmann) [inline docs/descriptor/includes/image/osbs.rst:62]: This block is unclear. You should wrap it into directory structure.

File: docs/descriptor/includes/image/osbs.rst [hunk 4]
```diff
@@ -50,6 +49,31 @@ in the :ref:`generation phase<handbook/building/build-process:Generating require
...
+    manifests
+    metadata
+
+will be copied to the root of the image filesystem, while with the below example a different target directory is chosen.
+
+.. code-block:: yaml
+
+    osbs:
+        extra_dir: custom-files
+        extra_dir_copy: /image-user/tmp
```
Comment (goldmann) [inline docs/descriptor/includes/image/osbs.rst:74]: What is `extra_dir_copy`?

File: docs/descriptor/includes/image/osbs.rst [hunk 5]
```diff
@@ -50,6 +49,31 @@ in the :ref:`generation phase<handbook/building/build-process:Generating require
...
 
+OSBS extra copy command
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Key
+    ``extra_dir_target``
+Required
+    No
+
+If this key is set then the contents of ``extra_dir`` will be copied during the image build to the location specified by this key. If this is *not* set, no ``COPY`` commands will be added. For example, with the value of ``/`` then
```
Comment (goldmann) [inline docs/descriptor/includes/image/osbs.rst:60]: Why not set the default value of the `extra_dir_target` to what is specified in `extra_dir_target` and copy it always to a directory with the same name and optionally changing the target name?

**Review by goldmann — CHANGES_REQUESTED** [review]

File: docs/descriptor/includes/image/osbs.rst (hunk 4 above)
Comment (rnc) [reply docs/descriptor/includes/image/osbs.rst:74]: That will teach me to write the docs before the code :-) Changing

**Review by goldmann — APPROVED** [review]

**Pull request merged**